File: src/main/java/engine/util/updater/ElapsedTimeUpdater.java
package engine.util.updater;


public class ElapsedTimeUpdater implements Updater {

    private UpdateFunc func;

    private final long updatePeriod;
    private long timeAccumulator;

    public ElapsedTimeUpdater(final long updatePeriod) {
        this.updatePeriod = updatePeriod;
        this.timeAccumulator = 0;
    }

    @Override
    public void setUpdateFunc(UpdateFunc func) {
        this.func = func;
    }

    @Override
    public void update() {
        ++this.timeAccumulator;
        execUpdateFuncOnCondition();
    }

    private boolean condition() {
        return timeAccumulator >= updatePeriod && func != null;
    }

    private void execUpdateFuncOnCondition() {
        if (condition()) func.exec();
    }
}

File: src/main/java/engine/util/updater/ElapsedTimeUpdater.kt
package engine.util.updater

class ElapsedTimeUpdater(private val updatePeriod: Long) : Updater {
    private var updateFunc: () -> Unit = {}
    private var timeAccumulator: Long = 0

    override fun setUpdateFunc(func: ()->Unit) {
        updateFunc = func
    }

    override fun update() {
        ++timeAccumulator
        execUpdateFuncOnCondition()
    }

    private fun condition(): Boolean {
        return timeAccumulator >= updatePeriod
    }

    private fun execUpdateFuncOnCondition() {
        if (condition()) updateFunc.invoke()
    }
}
